apiary-metastore-listener/src/main/java/com/expedia/apiary/extensions/metastore/listener/ApiarySnsListener.java

```diff
@@ -148,23 +150,31 @@ private void publishEvent(
     }
     if (partition != null) {
       JSONArray partitionValuesArray = new JSONArray(partition.getValues());
-      json.put("partition", partitionValuesArray);
+      LinkedHashMap<String, String> partitionKeysMap = new LinkedHashMap<>();
+      for (FieldSchema fieldSchema : partition.getSd().getCols()) {
+        partitionKeysMap.put(fieldSchema.getName(), fieldSchema.getType());
+      }
+
+      JSONObject partitionKeys = new JSONObject(partitionKeysMap);
+      json.put("partitionKeys", partitionKeys);
+      json.put("partitionValues", partitionValuesArray);
     }
     if (oldpartition != null) {
       JSONArray partitionValuesArray = new JSONArray(oldpartition.getValues());
-      json.put("oldPartition", partitionValuesArray);
+      json.put("oldPartitionValues", partitionValuesArray);
     }
 
     sendMessage(json);
   }
 
-  private void publishEvent(
+  private void publishInsertEvent(
       EventType eventType,
       String dbName,
       String tableName,
       Map<String, String> partitionKeyValues,
       List<String> files,
       List<String> fileChecksums) {
+
```

Comment:
Is this the blank line added to check whether I'm paying attention? ;)

Reply:
:) Removed